### ISSUE TITLE
Halyard not deploying the custom UDF template(s) in Clouddriver

### ISSUE THREAD

**Issue by iniinikoski** [spinnaker/spinnaker#4058]
### Issue Summary:

When trying to use the User Data Files-feature (https://www.spinnaker.io/setup/features/user-data/) with Spinnaker ( - Halyard does not deploy the files under clouddriver.

**Note:** this might be a duplicate of https://github.com/spinnaker/spinnaker/issues/3600. It has less details but this is possible...

### Cloud Provider(s):

--

### Environment:

Spinnaker version: 1.12.3 (latest as we speak)
Halyard version: 1.16.0 (latest as we speak)

Distributed setup on top of GKE.

### Feature Area (if this issue is UI/UX related, please tag @spinnaker/ui-ux-team):

Halyard.

### Description:

When custom UDF templates are added under Halyard's structure in Clouddriver - those are expected to be deployed to the setup. Now this is not happening.

**Note:** this might be a duplicate of https://github.com/spinnaker/spinnaker/issues/3600. It has less details but this is possible...

### Steps to Reproduce:

1. create a custom profile file into here: 

> ~/.hal/default/profiles/clouddriver/udf/udf0

2. apply changes with Halyard: 

> hal deploy apply --service-names clouddriver,monitoring-daemon

3. check if the template is found from clouddriver's hierarcy (= not found)

Checking from the Clouddriver-container:
```
bash-4.4$ pwd
/opt
bash-4.4$ find . -type f|grep udf0
bash-4.4$
```

### Additional Details:

--

**Comment by ncknt** [comment]
I've added https://github.com/spinnaker/halyard/pull/1230 which should fix Halyard picking up the udf files. You'd still need to pass the `udf.udfRoot: /opt/clouddriver/config/udf` and `udf.defaultLegacyUdf: false`. This could be ported to prior releases.

**Comment by iniinikoski** [comment]
Confirmed now: works like a charm with just-released Halyard version (1.17.0).
